libs/search-widget/src/widgets/popup-widget/PopupWidget.svelte

```diff
@@ -0,0 +1,138 @@
+<svelte:options
+  customElement="nuclia-popup"
+  accessors />
+
+<script lang="ts">
+  import { onMount } from 'svelte';
+  import SearchBar from '../search-widget/SearchBar.svelte';
+  import SearchResults from '../search-widget/SearchResults.svelte';
+  import { isEmptySearchQuery, pendingResults, showResults } from '../../core';
+  import type { KBStates } from '@nuclia/core';
+
+  export let backend = 'https://nuclia.cloud/api';
+  export let zone = 'europe-1';
+  export let knowledgebox: string;
+  export let placeholder = '';
+  export let lang = '';
+  export let cdn = '';
+  export let apikey = '';
+  export let kbslug = ''; // TODO: kbslug not needed anymore once regional system come into service
+  export let account = '';
+  export let client = 'widget';
+  export let state: KBStates = 'PUBLISHED';
+  export let features = '';
+  export let standalone = false;
+  export let proxy = false;
+  export let mode = '';
+  export let filters = '';
+  export let preselected_filters = '';
+  export let cssPath = '';
+  export let prompt = '';
+  export let generativemodel = '';
+  export let no_tracking = false;
+  export let rag_strategies = '';
+  export let rag_image_strategies = '';
+  export let not_enough_data_message = '';
+  export let ask_to_resource = '';
+
+  let searchBar: any;
+  let visible = false;
+
+  export function search(query: string) {
+    visible = true;
+    setTimeout(() => searchBar?.search(query), 0);
+  }
+
+  function enablePopupTrigger() {
+    const searchButtons = document.querySelectorAll('[data-nuclia="search-widget-button"]');
+    if (searchButtons.length > 0) {
+      searchButtons.forEach((button) => button.addEventListener('click', toggleSearchVisibility));
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  onMount(() => {
+    const firstTry = enablePopupTrigger();
+    if (!firstTry) {
+      setTimeout(() => {
+        const secondTry = enablePopupTrigger();
+        if (!secondTry) {
+          console.info(
+            `No button found to open Nuclia’s search widget. Make sure you added 'data-nuclia="search-widget-button"' to a clickable element.`,
```

Comment:
No button found to open Nuclia’s search widget. Make sure you added 'data-nuclia="search-widget-button"' **attribute** to a clickable element.

Reply:
debatable, the attribute is `data-nuclia` (and `"search-widget-button"` is its value), while `'data-nuclia="search-widget-button"'` is not an attribute, it's a just a snippet of code.